File: app/src/main/java/com/sqube/tipshub/ContactActivity.java
package com.sqube.tipshub;

import android.content.Intent;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Bundle;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

public class ContactActivity extends AppCompatActivity implements View.OnClickListener {
    PackageManager pkMgt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_contact);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle("");
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeAsUpIndicator(R.drawable.ic_close_black_24dp);
        }

        CardView crdWhatsapp = findViewById(R.id.crdWhatsapp);
        crdWhatsapp.setOnClickListener(this);
        CardView crdTwitter = findViewById(R.id.crdTwitter);
        crdTwitter.setOnClickListener(this);
        CardView crdFacebook = findViewById(R.id.crdFacebook);
        crdFacebook.setOnClickListener(this);
        CardView crdEmail = findViewById(R.id.crdEmail);
        crdEmail.setOnClickListener(this);

        pkMgt = getPackageManager();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        finish();
        return true;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.crdEmail:
                sendEmail();
                break;
            case R.id.crdWhatsapp:
                startWhatsapp();
                break;
            case R.id.crdTwitter:
                startBrowser("https://twitter.com/tipshub_co");
                break;
            case R.id.crdFacebook:
                startBrowser("https://web.facebook.com/Tipshub-110212083647469");
                break;
        }
    }

    private void sendEmail(){
        Intent sendIntent = new Intent(Intent.ACTION_SENDTO);
        sendIntent.setData(Uri.parse("mailto:swiftqube@gmail.com"));
        sendIntent.putExtra(Intent.EXTRA_SUBJECT, "Tipshub Help Request");
        startActivity(Intent.createChooser(sendIntent, "Select:"));
    }


    private void startWhatsapp() {
        String mssg = "Hello Tipshub";
        String toNumber = "2348132014755";
        Uri uri = Uri.parse("http://api.whatsapp.com/send?phone="+toNumber +"&text="+mssg);
        try {
            Intent whatsApp = new Intent(Intent.ACTION_VIEW);
            whatsApp.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY |
                    Intent.FLAG_ACTIVITY_MULTIPLE_TASK);
            whatsApp.setData(uri);
            pkMgt.getPackageInfo("com.whatsapp", PackageManager.GET_META_DATA);
            startActivity(whatsApp);
        }
        catch (PackageManager.NameNotFoundException e){
            Toast.makeText(this, "No WhatApp installed", Toast.LENGTH_LONG).show();
        }
    }

    private void startBrowser(String url) {
        Uri uri = Uri.parse(url);
        Intent intent = new Intent(Intent.ACTION_VIEW);
        intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY |
                Intent.FLAG_ACTIVITY_MULTIPLE_TASK);
        intent.setData(uri);
        startActivity(intent);
    }
}

File: app/src/main/java/com/sqube/tipshub/ContactActivity.kt
package com.sqube.tipshub

import android.content.Intent
import android.content.pm.PackageManager
import android.net.Uri
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView

class ContactActivity : AppCompatActivity(), View.OnClickListener {
    var pkMgt: PackageManager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_contact)
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.title = ""
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setHomeAsUpIndicator(R.drawable.ic_close_black_24dp)
        }
        val crdWhatsapp = findViewById<CardView>(R.id.crdWhatsapp)
        crdWhatsapp.setOnClickListener(this)
        val crdTwitter = findViewById<CardView>(R.id.crdTwitter)
        crdTwitter.setOnClickListener(this)
        val crdFacebook = findViewById<CardView>(R.id.crdFacebook)
        crdFacebook.setOnClickListener(this)
        val crdEmail = findViewById<CardView>(R.id.crdEmail)
        crdEmail.setOnClickListener(this)
        pkMgt = packageManager
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        finish()
        return true
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.crdEmail -> sendEmail()
            R.id.crdWhatsapp -> startWhatsapp()
            R.id.crdTwitter -> startBrowser("https://twitter.com/tipshub_co")
            R.id.crdFacebook -> startBrowser("https://web.facebook.com/Tipshub-110212083647469")
        }
    }

    private fun sendEmail() {
        val sendIntent = Intent(Intent.ACTION_SENDTO)
        sendIntent.data = Uri.parse("mailto:swiftqube@gmail.com")
        sendIntent.putExtra(Intent.EXTRA_SUBJECT, "Tipshub Help Request")
        startActivity(Intent.createChooser(sendIntent, "Select:"))
    }

    private fun startWhatsapp() {
        val mssg = "Hello Tipshub"
        val toNumber = "2349041463249"
        val uri = Uri.parse("http://api.whatsapp.com/send?phone=$toNumber&text=$mssg")
        try {
            val whatsApp = Intent(Intent.ACTION_VIEW)
            whatsApp.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY or
                    Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
            whatsApp.data = uri
            pkMgt!!.getPackageInfo("com.whatsapp", PackageManager.GET_META_DATA)
            startActivity(whatsApp)
        } catch (e: PackageManager.NameNotFoundException) {
            Toast.makeText(this, "No WhatApp installed", Toast.LENGTH_LONG).show()
        }
    }

    private fun startBrowser(url: String) {
        val uri = Uri.parse(url)
        val intent = Intent(Intent.ACTION_VIEW)
        intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY or
                Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
        intent.data = uri
        startActivity(intent)
    }
}